### ISSUE TITLE
A bug in Intersection of two Line3D?

### ISSUE THREAD

**Issue by zxo102** [sympy/sympy#8615]
I tested two perpendicular lines in three cases, Case 1: the lines are on x=6 plane,  Case 2: the lines are on xy plane (2D) and Case 3: the lines are on z=6 plane. Running the script below: Case 2 and Case 3 return correct results: Point(-1, 0)] and [Point3D(-1, 0, 6)], but Case 1 returns [] instead of  [Point3D(6, -1, 0)].  

Non-parallel 3D lines might not have a intersection point, but the following if-checking does not work for Case 1:
                ...  
                dx = a.x - b.x
                c = solve([dx, a.y - b.y], t)
                d = solve([dx, a.z - b.z], t)
                if len(c) == 1 and len(d) == 1:
                    return [] 
                ...

If commenting the above two lines, 
                ##if len(c) == 1 and len(d) == 1:
                ####return [] 

A correct intersection point is returned for Case 1 but this hacking is definitely not right.
## testing script #######################################################

from sympy import *
## case 1: two perpendicular lines on x=6 plane.

a=Line3D(Point3D(6, 5, 0), Point3D(6, -6, 0))
b=Line3D(Point3D(6, -1, 19/10), Point3D(6, -1, 0))
print(a.intersection(b))
## case 2: two perpendicular lines on xy plane.

c=Line(Point(5, 0), Point(-6, 0))
d=Line(Point(-1, 19/10), Point(-1, 0))
print(c.is_parallel(d))
print(c.intersection(d))
## case 3:  two perpendicular lines on z=6 plane.

c=Line3D(Point3D(5, 0,6), Point3D(-6, 0,6))
d=Line3D(Point3D(-1, 19/10,6), Point3D(-1, 0,6))
print(c.is_parallel(d))
print(c.intersection(d))
## line3d.py

  ...
  def intersection(self, o):
        ...
        elif isinstance(o, LinearEntity3D):
            if self == o:
                return [self]
            elif self.is_parallel(o):
               ...
            else:
                ...
                dx = a.x - b.x
                c = solve([dx, a.y - b.y], t)
                d = solve([dx, a.z - b.z], t)
                ##if len(c) == 1 and len(d) == 1:
                ###return []
                ...

ouyang

**Comment by namannimmo10** [comment]
@smichr @oscarbenjamin I think this issue has been fixed already.

```
>>> a = Line3D(Point3D(6,5,0), Point3D(6,-6,0))
>>> b = Line3D(Point3D(6,-1,19/10), Point3D(6,-1,0))
>>> a.intersection(b)
[Point3D(6, -1, 0)]
```
Should I add the required tests in a PR and close this issue?

**Comment by oscarbenjamin** [comment]
From `git bisect` I see that this was fixed in 2e29294da714. Does that commit already contain tests for this?

**Comment by namannimmo10** [comment]
I couldn't find any test related to Case 1 mentioned above. So, I've added that.
